File: Golang/01 Basic Algorithm/12 Mutations/mutations.go
package main

import (
	"fmt"
	"strings"
)

func mutation(slc, compare string) bool {
	//convert the case to lower for testing consistency
	target := strings.ToLower(slc)
	test := strings.ToLower(compare)
	//Loop through 'test'
	for _, v := range test {
		//Check if all the characters are present in 'target'
		if !strings.ContainsAny(target, string(v)) {
			return false
		}
	}
	return true
}

func main() {
	fmt.Println(mutation("Alien", "lIn"))
}

File: JS/01 Basic Algorithm/12 Mutations/mutations.js
function mutation(arr) {
    var target = arr[0].toLowerCase();
    var compare = arr[1].toLowerCase();
    for (var i = 0; i < compare.length; i++) {
        if (target.indexOf(compare[i]) === -1) {
            return false;
        }
    }
    return true;
}

mutation(["hello", "hey"]);
